### PR TITLE
Worker to automatically invoice pending platform fees and tips

### PR DESCRIPTION
**Pull request by kewitz**
Resolves https://github.com/opencollective/opencollective/issues/3650

### CODE REVIEW

**Comment by kewitz** [comment]
Rebased.

File: cron/monthly/invoice-platform-fees.js [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+    const host = await models.Collective.findByPk(hostId);
+    const connectedAccounts = await host.getConnectedAccounts({
+      where: { deletedAt: null },
+    });
+
+    let PayoutMethodId = FEES_ON_TOP_SETTLEMENT_EXPENSE_PROPERTIES.PayoutMethodId;
+    if (connectedAccounts?.find?.(c => c.service === 'transferwise')) {
+      PayoutMethodId = 2955;
+    } else if (connectedAccounts?.find?.(c => c.service === 'paypal') && !host.settings?.disablePaypalPayouts) {
+      PayoutMethodId = 6087;
```
Comment (znarf) [inline cron/monthly/invoice-platform-fees.js:178]: Can we avoid hardcoding here?

File: cron/monthly/invoice-platform-fees.js [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    // Create Expense Items
+    items = items.map(i => ({
+      ...i,
+      ExpenseId: expense.id,
+      CreatedByUserId: FEES_ON_TOP_SETTLEMENT_EXPENSE_PROPERTIES.UserId,
+    }));
+    await models.ExpenseItem.bulkCreate(items);
+
+    // Attach CSV
+    const Body = json2csv(hostTransactions.map(t => omit(t, ['data'])));
```
Comment (znarf) [inline cron/monthly/invoice-platform-fees.js:204]: I found the CSV sample not ideal in the choice of columns featured. Can we consider something more compact?

File: cron/monthly/invoice-platform-fees.js [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+      } (${currency})`,
+    );
+
+    // Credit the Host with platform tips collected during the month
+    await models.Transaction.create({
+      amount: totalAmount,
+      amountInHostCurrency: totalAmount,
+      CollectiveId: hostId,
+      // Pia's account
+      CreatedByUserId: 30,
```
Comment (znarf) [inline cron/monthly/invoice-platform-fees.js:159]: That could maybe be a constant or a configuration variable.

**Pull request merged**